src/Monad/Monad.lidr

```diff
@@ -0,0 +1,86 @@
+\iffalse
+SPDX-License-Identifier: AGPL-3.0-only
+
+This file is part of `idris-ct` Category Theory in Idris library.
+
+Copyright (C) 2019 Stichting Statebox <https://statebox.nl>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+\fi
+
+> module Monad.Monad
+>
+> import Basic.Category
+> import Basic.Functor
+> import Basic.NaturalTransformation
+> import Cats.CatsAsCategory
+> import CategoryReasoning as CR
+> import Syntax.PreorderReasoning
+> import Functors.FunctorCategories
+>
+> %access public export
+> %default total
+>
+> qed : (a : obj (functorCategory cat cat)) -> mor _ a a
+> qed = CR.qed (functorCategory cat cat)
+>
+> step : (a : obj (functorCategory cat cat)) -> mor (functorCategory cat cat) a b -> mor (functorCategory cat cat) b c -> mor (functorCategory cat cat) a c
+> step = CR.step (functorCategory cat cat)
+>
+> liftEquality : (a, b : obj (functorCategory cat cat)) -> a = b -> mor (functorCategory cat cat) a b
+> liftEquality = CR.liftEquality (functorCategory cat cat)
+>
+> record Monad (cat : Category) where
+>   constructor MkMonad
+>   functor : CFunctor cat cat
+>   unit : NaturalTransformation _ _ (idFunctor cat) functor
+>   multiplication : NaturalTransformation cat cat (functorComposition _ _ _ functor functor) functor
+>   associativity :
```

Comment:
maybe it could be nice to have these properties proved on separate lemmas, to make the actual definition a bit more concise, what do you say?

Reply:
The issue is that these are not lemmas, these are commutative diagrams (of natural transformations). `associativity` for example is translated from the following agda code:
```
associativity :
  (begin
    F ∘ (F ∘ F)
  -⟨ μ N∘F F ⟩→
    F ∘ F
  -⟨ μ ⟩→
    F
  ∎) ≡
  (begin
    F ∘ (F ∘ F) ≡⟨ functorAssociativity F F F ⟩ (F ∘ F) ∘ F
  -⟨ F F∘N μ ⟩→
    F ∘ F
  -⟨ μ ⟩→
    F
  ∎)
```
If you have some ideas of how to improve the level of readability I'm all for it

Reply:
You're right, these are types. I meant something like what was done for the `Associator`, the `MonoidalPentagon` or the `MonoidalTriangle` axioms of a `Monoidalcategory` (https://github.com/statebox/idris-ct/blob/e21c78e6fc7bc564f951c907e1eff79e4041e1b5/src/MonoidalCategory/MonoidalCategory.lidr#L34).

Could be something like

```
MonadAssociativity : Type
MonadAssociativity = ...
```

I'm not sure if this is actually a good idea, but I'm wondering if it could improve readability

Reply:
I'm not sure, it might make the file a bit cleaner, but at the same time it splits something off that doesn't really have a use elsewhere... Both of these options don't make me happy personally...